src/routes/fetch-posts.e2e.test.js

```diff
@@ -0,0 +1,181 @@
+			const nonAdminUser = await fixtures.insertUser({
+				role : 'user'
+			}) 
+			const nonAdminAccessToken = fixtures.generateAccessToken(
+				{user :  { id: nonAdminUser.id, email: nonAdminUser.email }}
+			)
+			const expectedError = {
+				"status":"failed",
+				"message":"Only admin can create users"
+			}
+			return fixtures.api()
+				.get(`/api/v1/feed?isFlagged=${true}`)
+				.set('Authorization', `Bearer ${nonAdminAccessToken}`)
+				.expect(403 , expectedError)
+		
+		}) 
+
+	})
+	
+	describe('Success', () => {
+
+		const numberOfPosts = 3
+		let insertedPosts
+
+		beforeEach(async () =>{
+
+			await resetDBTable('posts')
+			await resetDBTable('post_flags')
+			insertedPosts= await fixtures.insertMultiplePosts(
+				user.id,
+				numberOfPosts
+			)
+			
+		})
+
+		it('should return 200 if posts are fetched', async () =>
+			
+			fixtures.api()
+				.get(`/api/v1/feed`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.expect(200)
+		)
+
+		it('should return the right number of posts', async() =>
+			
+			fixtures.api()
+				.get(`/api/v1/feed`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.then(res => {
+
+					expect(res.body.data).to.have.deep.members(
+
+						insertedPosts.map((post) =>({
+							createdOn: post.createdAt.toISOString(),
+							gifId : post.id,
+							imageUrl: post.content,
+							likesCount : post.likesCount,
+							published : post.published,
+							title : post.title,
+							user: {
+								userId: user.id,
+								fullName: `${user.firstName} ${user.lastName}`,
+								profilePictureUrl : user.profilePictureUrl,
+								email : user.email
+							}
+							
+						})))
+				}) 
+		)	
+
+		it('should return flagged posts', async() =>{
+			const post = insertedPosts[0]
+			await fixtures.insertPostFlag({
+				userId : user.id ,
+				postId : post.id
+			})
+		
+
+			return fixtures.api()
+				.get(`/api/v1/feed?isFlagged=${true}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.then(res => {
+					
+					expect(res.body.data[0]).to.eql({
+	
+						createdOn: post.createdAt.toISOString(),
+						gifId : post.id,
+						imageUrl: post.content,
+						flagsCount : 1,
+						likesCount : post.likesCount,
+						published : post.published,
+						title : post.title,
+						user: {
+							userId: user.id,
+							fullName: `${user.firstName} ${user.lastName}`,
+							profilePictureUrl : user.profilePictureUrl,
+							email : user.email
+						}
+						
+							
+					})
+				}) 
+		})
+
+		it('should return unflagged posts', async() =>{
+			
+			const insertedPost = insertedPosts[2]
+			await fixtures.insertPostFlag({
+				userId : user.id ,
+				postId : insertedPost.id
+			})
+		
+
+			return fixtures.api()
+				.get(`/api/v1/feed?isFlagged=${false}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.then(res => {
+					
+					expect(res.body.data).to.have.deep.members(
+	
+						insertedPosts
+							.slice(0,2)
```

Comment:
test flow doesn't make it clear why we need to pick the first 2. its not too clear and readable

Reply:
we pick the first two because we flagged the posts with an id of 2 hence the unflagged ones are the ones with the id of 0 and 1

Reply:
okay, that means a new person reason has to know there's 3 posts being inserted, the 3rd one was flagged and the unflagged ones are these ones

at least comments and variable naming helps one to follow without having to track stuff in the test to understand what is going on

for example
1. since its the last post anyway, the flagging of the '3rd' post can simply be 

```js
  lastPost = insertedPosts[insertedPosts.length -1)

 // flag only last post
 await insertPostFlag({
   ......
 })
```


2.  the `insertedPosts.slice(0,2)` can be assigned into a variable `unflaggedPosts` before being mapped over. and based on the change in step 1, can also be updated to `insertedPosts.slice(0, insertedPosts.length - 1)`

Reply:
Oh okay, I agree with you, this will make it extremely easy for a new person to track. Thanks a lot

Reply:
i have implemented and pushed all suggested changes, kindly review